crates/services/block_aggregator_api/src/blocks/importer_and_db_source.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{
+    blocks::{
+        Block,
+        BlockSource,
+        BlockSourceEvent,
+        importer_and_db_source::importer_service::ImporterTask,
+    },
+    result::{
+        Error,
+        Result,
+    },
+};
+use anyhow::anyhow;
+use fuel_core_services::{
+    Service,
+    ServiceRunner,
+    stream::BoxStream,
+};
+use fuel_core_storage::{
+    StorageInspect,
+    tables::FuelBlocks,
+};
+use fuel_core_types::{
+    blockchain::block::Block as FuelBlock,
+    fuel_types::BlockHeight,
+    services::block_importer::SharedImportResult,
+};
+
+use crate::blocks::importer_and_db_source::sync_service::SyncTask;
+use fuel_core_storage::tables::Transactions;
+
+pub mod importer_service;
+pub mod sync_service;
+#[cfg(test)]
+mod tests;
+
+pub trait BlockSerializer {
+    fn serialize_block(&self, block: &FuelBlock) -> Result<Block>;
+}
+
+pub struct ImporterAndDbSource<Serializer, DB>
+where
+    Serializer: BlockSerializer + Send + Sync + 'static,
+    DB: Send + Sync + 'static,
+    DB: StorageInspect<FuelBlocks>,
+    DB: StorageInspect<Transactions>,
+    <DB as StorageInspect<FuelBlocks>>::Error: std::fmt::Debug + Send,
+{
+    // TODO: How to handle errors from these tasks?
+    _importer_task: ServiceRunner<ImporterTask<Serializer>>,
+    _sync_task: ServiceRunner<SyncTask<Serializer, DB>>,
+    receiver: tokio::sync::mpsc::Receiver<BlockSourceEvent>,
+}
+
+impl<Serializer, DB> ImporterAndDbSource<Serializer, DB>
+where
+    Serializer: BlockSerializer + Clone + Send + Sync + 'static,
+    DB: StorageInspect<FuelBlocks> + Send + Sync,
+    DB: StorageInspect<Transactions> + Send + 'static,
+    <DB as StorageInspect<FuelBlocks>>::Error: std::fmt::Debug + Send,
+{
+    pub fn new(
+        importer: BoxStream<SharedImportResult>,
+        serializer: Serializer,
+        database: DB,
+        db_starting_height: BlockHeight,
+        db_ending_height: BlockHeight,
+    ) -> Self {
+        const ARB_CHANNEL_SIZE: usize = 100;
```

Comment:
what happens to the import task if the service consuming the `BlockSource` is slow?